packages/medusa/src/services/shipping-profile.ts

```diff
@@ -352,43 +354,62 @@ class ShippingProfileService extends TransactionBaseService {
   }
 
   /**
-   * Adds a product to a profile. The method is idempotent, so multiple calls
-   * with the same product variant will have the same result.
-   * @param profileId - the profile to add the product to.
-   * @param productId - the product to add.
+   * Adds an array of products to the profile.
+   * @param profileId - the profile to add the products to.
+   * @param productIds - the products to add.
    * @return the result of update
    */
-  async addProduct(
+  async addProducts(
     profileId: string,
-    productId: string
+    productIds: string[]
   ): Promise<ShippingProfile> {
     return await this.atomicPhase_(async (manager) => {
-      await this.productService_
-        .withTransaction(manager)
-        .update(productId, { profile_id: profileId })
+      const productServiceTx = this.productService_.withTransaction(manager)
+      for (const pId of productIds) {
+        await productServiceTx.update(pId, {
+          profile_id: profileId,
+        })
+      }
 
-      return await this.retrieve(profileId)
+      return await this.retrieve(profileId, {
+        relations: [
+          "products",
+          "products.profile",
+          "shipping_options",
+          "shipping_options.profile",
+        ],
+      })
     })
   }
 
   /**
    * Adds a shipping option to the profile. The shipping option can be used to
    * fulfill the products in the products field.
    * @param profileId - the profile to apply the shipping option to
-   * @param optionId - the option to add to the profile
+   * @param optionIds - the option to add to the profile
    * @return the result of the model update operation
    */
-  async addShippingOption(
+  async addShippingOptions(
     profileId: string,
-    optionId: string
+    optionIds: string[]
```

Comment:
thought: This breaks the API. Should we add `optionIds: string | string[]` to remain backward compatible and keep the name? We use that pattern extensively already. 

Would require a check like:
```
if (Array.isArray(optionsIds) {
...
```